server/src/test/java/org/opensearch/telemetry/TestInMemoryMetricsRegistry.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry;
+
+import org.opensearch.telemetry.metrics.Counter;
+import org.opensearch.telemetry.metrics.Histogram;
+import org.opensearch.telemetry.metrics.MetricsRegistry;
+import org.opensearch.telemetry.metrics.tags.Tags;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Supplier;
+
+/**
+ * This is a simple implementation of MetricsRegistry which can be utilized by Unit Tests.
+ * It just initializes and stores counters/histograms within a map, once created.
+ * The maps can then be used to get the counters/histograms by their names.
+ */
+public class TestInMemoryMetricsRegistry implements MetricsRegistry {
+
+    private ConcurrentHashMap<String, TestInMemoryCounter> counterStore;
+    private ConcurrentHashMap<String, TestInMemoryHistogram> histogramStore;
+
+    public TestInMemoryMetricsRegistry() {
+        this.counterStore = new ConcurrentHashMap<>();
+        this.histogramStore = new ConcurrentHashMap<>();
```

Comment:
nit can be move to inline - and remove constructor.

Reply:
Updated in the revision.